src/routes/transactions/mappers/common/native-staking.mapper.ts

```diff
@@ -283,29 +277,34 @@ export class NativeStakingMapper {
    * and after execution it returns 0. Therefore, if the transaction was executed
    * we return the value get the exact value from the transaction logs instead.
    *
-   * @param {MultisigTransaction | ModuleTransaction | null} args.transaction - the `batchWithdrawCLFee` transaction
-   * @param {DataDecoded} args.dataDecoded - the decoded data of the transaction
+   * @param {string | nulle} args.txHash - the transaction hash of the withdraw transaction
+   * @param {Array<`0x${string}`>} args.publicKeys - the public keys to get the value for
    * @param {string} args.chainId - the chain ID of the native staking deployment
+   * @param {string} args.safeAddress - the Safe staking
    *
    * @returns {number} the value to withdraw or withdrawn
    */
   private async getWithdrawValue(args: {
-    transaction: MultisigTransaction | ModuleTransaction | null;
-    dataDecoded: DataDecoded;
+    txHash: `0x${string}` | null;
+    publicKeys: Array<`0x${string}`>;
     chainId: string;
     safeAddress: `0x${string}`;
   }): Promise<number> {
-    if (!args.transaction?.transactionHash) {
-      return this.kilnNativeStakingHelper.getValueFromDataDecoded({
+    if (!args.txHash) {
+      const stakes = await this.stakingRepository.getStakes({
         chainId: args.chainId,
         safeAddress: args.safeAddress,
-        dataDecoded: args.dataDecoded,
+        validatorsPublicKeys: args.publicKeys,
       });
+      return stakes.reduce((acc, stake) => {
+        const netValue = stake.net_claimable_consensus_rewards ?? '0';
+        return acc + Number(netValue);
+      }, 0);
     }
```

Comment:
Nit: I'm OK with changing the way the withdraw value is retrieved/calculated in this PR, but it seems to me a relevant/big change. I'm under the impression this is not tied to the Confirmation View deprecation so what do you think about saying something about the decoding of the staking transaction data (instead of using dataDecoded) in the PR title?

Reply:
After discussion, we've agreed to keep this as is. The preview endpoint does not fetch decoded data in a similar way to the native staking mapper, where we assign fallback values for Kiln contracts. As such, this change is necessary here.